### ISSUE TITLE
NVT updates won't finish properly in case openvas scan is finished, but still running for gsad / gvmd and so will stuck new scans requests

### ISSUE THREAD

**Issue by wisukind** [greenbone/ospd-openvas#298]
**Environment:**
```
OpenVAS 7.0.1
gvm-libs 11.0.1
OSP Server for openvas: 1.0.1
OSP: 1.2
OSPd: 2.0.1
python2.6
Ubuntu 18.04 LTS
Redis 4.09 with GVMd tuned configuration file

```

**Current behaviour:**
If you have a scan shown as completed on openvas / ospd-openvas side, but still in Running state on gvmd / gsad side, and a NVT update start; then the update won't finish properly preventing new scans to start.

In the following example; scan has terminated properly and later a NVT update is launched:
```
sd   main:MESSAGE:2020-07-19 19h39.45 utc:32689: Total time to scan all hosts : 49142 seconds
lib  nvticache:MESSAGE:2020-07-24 09h15.32 utc:5936: Updated NVT cache from version 202007071016 to 202007230955
```
On ospd-openvas side; scan is seen has finished but update never finish:
```
2020-07-23 04:38:48,951 OSPD - openvas: INFO: (ospd.ospd) 68cab482-415c-4475-9f9b-d2dcd6f562b3: Scan finished.
2020-07-24 11:15:10,793 OSPD - openvas: INFO: (ospd_openvas.daemon) A feed update process is running. Trying again later...
2020-07-24 11:15:10,793 OSPD - openvas: DEBUG: (ospd_openvas.daemon) The feed was not upload or it is outdated, but other process is locking the update. Trying again later...
2020-07-24 11:15:20,857 OSPD - openvas: DEBUG: (ospd_openvas.daemon) Loading NVTs in Redis DB
2020-07-24 11:15:32,836 OSPD - openvas: DEBUG: (ospd_openvas.daemon) Feed lock file removed.
```

However; the update is still considered as pending, even hours after last logged event, and the following update log never show up:
`OSPD - openvas: INFO: (ospd_openvas.daemon) Finish loading up vts.`

As a consequence, further scan requests will automatically fails since ospd-openvas still consider there is a pending update:
`2020-07-25 10:56:50,170 OSPD - openvas: INFO: (ospd_openvas.daemon) c6ccea2a-3ca3-47a2-be89-d6319fb8f9d3: There is a pending feed update. The scan can not be started.`

**Expected behaviour:**
When a scan is done on openvas / ospd-openvas side; NVT updates shouldn't be stuck and should finish properly.

**How to reproduce:**

1. Start a scan on somehow important tasks, so that results calculation by gvmd will keep running after ospd-openvas complete
2. When openvas / ospd-openvas side has completed task scan; run a feed update on the scanner 
3. After a reasonable delay, where feed update should be finished, launch a new scan
4. Scan will be automatically set in "Done" state with report set in "Error" as a pending update is stuck forever.

**Comment by jjnicola** [comment]
Hi @wisukind,
this can be related to greenbone/openvas#525, and given the case, it is already solved with geenbone/openvas#527.
Regards,

**Comment by jjnicola** [comment]
Version 20.08 includes a scan queue system which helps to address many issues related to started/running scans and the feed update. Also, vts load up process was improved, since ospd-openvas doesn't use the internal VT storage, but a direct connection to redis. These improvements will not be backported to ospd-openvas v1.0. Closing the issue here.
